File: ESC.Resturant.Data/Entities/Bill.cs
using ESA.Data;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ESC.Resturant.Data.Entities
{
    public class Bill
    {

        [Key]
        public int Id { get; set; }
        public string Code { get; set; }
        public Nullable<int> OrderNo { get; set; }
        public string TableNo { get; set; }
        public Nullable<int> PeopleID { get; set; }
        public string PepoleName { get; set; }
        public Nullable<System.DateTime> CurrentDate { get; set; }
        public Nullable<System.DateTime> Date { get; set; }
        public Nullable<double> TotalQty { get; set; }
        public Nullable<double> SupTotal { get; set; }
        public Nullable<double> TotalVatTax { get; set; }
        public Nullable<double> TotalAfterVatTax { get; set; }
        public Nullable<double> TotalNotePrice { get; set; }
        public Nullable<double> CurrentDiscount { get; set; }
        public Nullable<double> DescountValue { get; set; }
        public Nullable<double> ServiceFees { set; get; }
        public Nullable<double> ServiceFeesValue { set; get; }
        public Nullable<double> ApplicationValue { get; set; }
        public Nullable<double> NetTotal { get; set; }
        public Nullable<double> Paied { get; set; }
        public Nullable<double> Remaining { get; set; }
        public string Notes { get; set; }
        public string Reference { get; set; }
        public Nullable<int> EmployeeId { get; set; }
        public string UserId { get; set; }
        public Nullable<int> BillTypeId { get; set; }
        public Nullable<int> BillPlaceId { get; set; }
        public Nullable<int> PaymentId { get; set; }
        public Nullable<int> VisaID { get; set; }
        public Nullable<int> ApplicationId { get; set; }
        public Nullable<int> ApplicationDiscountId { get; set; }
        public Nullable<int> DiscountId { get; set; }
        public Nullable<int> CurrencyId { get; set; }
        public Nullable<bool> CheckWiteInvoies { get; set; }
        public Nullable<bool> IsDelete { get; set; }
        public Nullable<int> IsApproverd { get; set; }
        public Nullable<int> BranchId { get; set; }




        public virtual ICollection<BillDetails> BillDetails { get; set; }
        public virtual ICollection<BillTaxes> BillTaxes { get; set; }
        public virtual ICollection<BillCurrencies> BillCurrencies { get; set; }
        public virtual ICollection<BillDeliveries> BillDeliveries { get; set; }

        public virtual BillPlace BillPlace { get; set; }
        public virtual Application Application { get; set; }
        public virtual DiscountType DiscountType { get; set; }
        public virtual Currencies Currencies { get; set; }
        public virtual BillType BillType { get; set; }
        public virtual PaymentType PaymentType { get; set; }
        public virtual Branch Branch { get; set; }
        public virtual ApplicationUser User { get; set; }



    }
}

File: ESAResturant.NEW.web/ClientApp/src/app/admin/bill-invoice/Models/Bill.ts
import { BillDetail } from './BillDetail';
import { BillPlace } from './BillPlace';
import { BillType } from './BillType';
import { PaymentType } from './PaymentType ';
import { Application } from '../../setting/Models/Application';
import { BillTaxes } from './BillTaxes';
import { BillCurrencies } from './BillCurrencies';
import { Currencies } from '../../setting/Models/Currencies';
import { DiscountType } from '../../setting/Models/DiscountType';
import { BillDeliveries } from './BillDeliveries';
import { People } from '../../People/Models/people';
import { CompanyInfo } from 'src/app/users/Models/CompanyInfo';
import { Branch } from '../../setting/Models/Branch';
import { User } from '../../../users/Models/User';

export class Bill {

  constructor() {
    this.id = 0;
    this.code = '';
    this.orderNo = 0;
    this.billDetails = [];
    this.billTaxes = [];
    this.billCurrencies = [];
    this.billDeliveries = [];

    this.tableNo = null;

    this.peopleID = 0;
    this.totalQty = 0;
    this.supTotal = 0;
    this.totalVatTax = 0;
    this.totalAfterVatTax = 0;

    this.currentDiscount = 0;
    this.descountValue = 0;
    this.serviceFees = 0;
    this.serviceFeesValue = 0;
    this.totalNotePrice = 0;

    this.netTotal = 0;
    this.paied = 0;
    this.applicationValue = 0;
    this.remaining = 0;
    this.employeeId = 0;
    //this.branchId = 0;
    this.billTypeId = 0;
    this.billPlaceId = 1;
    this.paymentId = 1;
    this.visaID = 0;
    this.userId = '';

    //this.applicationId = null;
    this.discountId = 1;
    this.currencyId = 0;
    this.applicationDiscountid = 0;

    this.checkWiteInvoies = false;
    this.isDelete = false;
    this.isApproverd = 0;
    this.currentLang = '';
    this.reference = '';
    this.notes = '';


    this.billPlace = new BillPlace();
    this.billType = new BillType();
    this.paymentType = new PaymentType();
    this.application = new Application();
    this.currencies = new Currencies();
    this.discountType = new DiscountType();
    this.branch = new Branch();
    this.user = new User();
    this.companyInfo = new CompanyInfo();

  }

  id: number;
  code: string;
  orderNo: number;
  tableNo: string;
  peopleID: number;
  pepoleName: string;
  currentDate: any;
  date: any;
  totalQty: number;
  supTotal: number;
  totalVatTax: number;
  totalAfterVatTax: number;
  totalNotePrice: number;

  currentDiscount: number;
  descountValue: number;
  serviceFees: number;
  serviceFeesValue: number;

  applicationValue: number;

  netTotal: number;
  paied: number;
  remaining: number;
  applicationDiscountid: number;

  notes: string;
  reference: string;
  employeeId: number;
  branchId: number;
  userId: string;
  billTypeId: number;
  billPlaceId: number;
  paymentId: number;
  discountId: number;
  visaID: number;
  applicationId: number;
  currencyId: number;
  checkWiteInvoies: boolean;
  isDelete: boolean;
  isApproverd?: number;
  currentLang: string;
  //optional


  billDetails: BillDetail[];
  billTaxes: BillTaxes[];
  billCurrencies: BillCurrencies[];
  billDeliveries: BillDeliveries[];

  billPlace: BillPlace;
  billType: BillType;
  paymentType: PaymentType;
  application: Application
  currencies: Currencies;
  discountType: DiscountType;
  branch: Branch;
  user: User;
  companyInfo: CompanyInfo;
}

